scripts/publish.sh

```diff
@@ -0,0 +1,332 @@
+        exit 3
+    fi
+}
+
+check_npm_stuff_is_stable () {
+    for pkgdir in ${PACKAGE_DIRS[@]}; do
+        pkgname="$(npm_pkgname "$pkgdir")"
+        echo "Rebuilding node_modules for $pkgname (this may take a while)..."
+        ( cd "$pkgdir" && (
+            # Before bumping anything, first make sure that all projects have
+            # a clean and stable node_modules directory and lock files!
+            rm -rf node_modules
+
+            logfile="$(mktemp)"
+            if ! npm install > "$logfile" 2> "$logfile"; then
+                cat "$logfile" >&2
+                err ""
+                err "The error above happened during the building of $PKGDIR."
+                exit 4
+            fi
+
+            if git is-dirty; then
+                err "I just removed node_modules and reinstalled all package dependencies"
+                err "inside $pkgdir, and found unexpected changes in the following files:"
+                err ""
+                ( cd "$ROOT" && git modified )
+                err ""
+                err "Please fix those issues first."
+                exit 2
+            fi
+        ) )
+    done
+}
+
+check_all_the_things () {
+    if [ -n "$VERSION" ] && ! is_valid_version "$VERSION"; then
+        # Check for typos early on
+        err "Invalid version: $VERSION"
+        exit 2
+    fi
+
+    check_git_toolbelt_installed
+    check_jq_installed
+    check_moreutils_installed
+    check_current_branch
+    check_up_to_date_with_upstream
+    check_cwd
+    check_no_local_changes
+    check_npm_stuff_is_stable
+}
+
+check_all_the_things
+
+CURRENT_VERSION="$(jq -r .version "$PRIMARY_PKG/package.json")"
+if [ -z "$VERSION" ]; then
+    echo "The current version is: $CURRENT_VERSION"
+fi
+
+while ! is_valid_version "$VERSION"; do
+    if [ -n "$VERSION" ]; then
+        err "Invalid version number: $VERSION"
+        err "Please try again."
+        err ""
+    fi
+    read -p "Enter a new version: " VERSION
+done
+
+bump_version_in_pkg () {
+    PKGDIR="$1"
+    VERSION="$2"
+
+    jq ".version=\"$VERSION\"" package.json | sponge package.json
+
+    # If this is one of the client packages, also bump the peer dependency
+    if [ "$(jq '.peerDependencies."@liveblocks/client"' package.json)" != "null" ]; then
+        jq ".peerDependencies.\"@liveblocks/client\"=\"$VERSION\"" package.json | sponge package.json
+    fi
+
+    prettier --write package.json
+
+    logfile="$(mktemp)"
+    if ! npm install > "$logfile" 2> "$logfile"; then
+        cat "$logfile" >&2
+        err ""
+        err "The error above happened during the building of $PKGDIR."
+        exit 4
+    fi
+
+    if [ "$CURRENT_VERSION" != "$VERSION"; then
+        if ! git modified | grep -qEe package-lock.json; then
+            err "Hmm. package-lock.json wasn\'t affected by the version bump. This is fishy. Please manually inspect!"
+            exit 5
+        fi
+    fi
+}
+
+build_pkg () {
+    rm -rf lib
```

Comment:
We're cleaning `lib` in every `build` scripts so I think we can make the assumptions that every build script should clean every previously built artifacts.

Reply:
Fantastic. Updated in 1fc1bb442bd49a74748ff617c5f76454c306dc2c.